src/ast/mod.rs

```diff
@@ -217,6 +217,18 @@ pub enum Expr {
     Identifier(Ident),
     /// Multi-part identifier, e.g. `table_alias.column` or `schema.table.col`
     CompoundIdentifier(Vec<Ident>),
+    /// Multi-part identifier with a column number, e.g. [alias.]$file_col_num[.element] (snowflake)
+    CompoundIdentifierWithColumnNumber {
```

Comment:
In terms of supporting things like `a.$1` -- what would you think about modeling this as a a two part `CompoundIdent` identifier `["a", "$1"]` rather than a new variant?

I think the parser would still have to be changed, but I think that approach would keep the AST and the consumption of the AST smaller with fewer dialect specific structures